app/src/main/java/dev/arkbuilders/arkmemo/ui/fragments/EditTextNotesFragment.kt

```diff
@@ -28,9 +32,23 @@ class EditTextNotesFragment: BaseEditNoteFragment() {
         requireContext().getTextFromClipBoard(view) { clipBoardText ->
             if (clipBoardText != null) {
                 val newTextBuilder = StringBuilder()
-                newTextBuilder.append(binding.editNote.text.toString()).append(clipBoardText)
+                val cursorPos = binding.editNote.selectionStart
+                val noteContent = binding.editNote.text.toString()
+
+                val newCursorPos = if (cursorPos < 0) {
+                    (clipBoardText.length + noteContent.length - 1).coerceAtLeast(0)
+                } else {
+                    clipBoardText.length + cursorPos
+                }
+                try {
+                    newTextBuilder.append(noteContent.insertStringAtPosition(clipBoardText, cursorPos))
+                } catch (e: IndexOutOfBoundsException) {
+                    Log.e(TAG, "pasteNoteClickListener exception: ${e.message}")
+                    newTextBuilder.append(noteContent).append(clipBoardText)
```

Comment:
Does this mean, pasting is not always dependent on cursor position?

Reply:
@shubertm In theory pasting should be dependent on cursor position, but if there's a case where the cursor position is invalid (e.g, cursor is not found anywhere within the text), then the pasted content should be appended to the end of the note's text.